srcs/python/quiver/partition.py

```diff
@@ -1,127 +1,139 @@
 import torch
+from quiver.utils import cal_memory_budget_to_bytes
+import shutil
+import os
 
-CHUNK_NUM = 32
 
+__all__ = ["quiver_partition"]
+QUIVER_MAGIC_NUMBER = 64
 
-def partition_without_replication(device, probs, ids):
-    """Partition node with given node IDs and node access distribution.
+def partition_without_replication(probs):
+    """Partition node with node access distribution. 
     The result will cause no replication between each parititon.
-    We assume node IDs can be placed in the given device.
 
     Args:
-        device (int): device which computes the partitioning strategy
         probs (torch.Tensor): node access distribution
-        ids (Optional[torch.Tensor]): specified node IDs
 
     Returns:
         [torch.Tensor]: list of IDs for each partition
 
     """
-    ranks = len(probs)
-    if ids is not None:
-        ids = ids.to(device)
-    probs = [
-        prob[ids].to(device) if ids is not None else prob.to(device)
-        for prob in probs
-    ]
-    total_size = ids.size(0) if ids is not None else probs[0].size(0)
-    res = [None] * ranks
-    for rank in range(ranks):
-        res[rank] = []
-    CHUNK_SIZE = (total_size + CHUNK_NUM - 1) // CHUNK_NUM
-    chunk_beg = 0
-    beg_rank = 0
-    for i in range(CHUNK_NUM):
-        chunk_end = min(total_size, chunk_beg + CHUNK_SIZE)
-        chunk_size = chunk_end - chunk_beg
-        chunk = torch.arange(chunk_beg,
-                             chunk_end,
-                             dtype=torch.int64,
-                             device=device)
+
+    device = torch.cuda.current_device()
+    partitioned_num = len(probs)
+
+    probs = [prob.to(device) for prob in probs]
+    total_node_num = probs[0].size(0)
+
+    res = [[] for _ in range(partitioned_num)]
+
+    chunk_size = QUIVER_MAGIC_NUMBER * partitioned_num
+    chunk_num = (total_node_num + chunk_size - 1) // chunk_size
```

Comment:
I think this number could be used as an argument but with default value:
`partition_without_replication(probs, chunk_num=64)`

Reply:
okk~